File: ts/src/journey/resources/apps.ts
import { HttpTransport } from "../../core/http.js"

export class AppsResource {
  constructor(private readonly http: HttpTransport, private readonly base: string) {}

  private get url() { return `${this.base}/journeys/v2/apps` }

  async submit(appId: string, body: Record<string, unknown>) {
    return this.http.getFetch()(`${this.url}/submit/${appId}`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(body),
    }).then(r => r.json())
  }

  async updateSettings(appId: string, body: Record<string, unknown>) {
    return this.http.getFetch()(`${this.url}/settings/${appId}`, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(body),
    }).then(r => r.json())
  }
}

File: py/src/imbrace/journey/resources/apps.py
from typing import Any, Dict
from ...core.http import HttpTransport, AsyncHttpTransport


class AppsResource:
    """Apps — Journey API. Submit + Settings.

    Endpoints:
        POST /journeys/v2/apps/submit/{app_id}
        PUT  /journeys/v2/apps/settings/{app_id}
    """

    def __init__(self, http: HttpTransport, base: str):
        self._http = http
        self._base = f"{base}/journeys/v2/apps"

    def submit(self, app_id: str, body: Dict[str, Any]) -> Dict[str, Any]:
        return self._http.request("POST", f"{self._base}/submit/{app_id}", json=body).json()

    def update_settings(self, app_id: str, body: Dict[str, Any]) -> Dict[str, Any]:
        return self._http.request("PUT", f"{self._base}/settings/{app_id}", json=body).json()


class AsyncAppsResource:
    """Apps — Journey API. Async."""

    def __init__(self, http: AsyncHttpTransport, base: str):
        self._http = http
        self._base = f"{base}/journeys/v2/apps"

    async def submit(self, app_id: str, body: Dict[str, Any]) -> Dict[str, Any]:
        res = await self._http.request("POST", f"{self._base}/submit/{app_id}", json=body)
        return res.json()

    async def update_settings(self, app_id: str, body: Dict[str, Any]) -> Dict[str, Any]:
        res = await self._http.request("PUT", f"{self._base}/settings/{app_id}", json=body)
        return res.json()
